### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by codificat**
Simplify jenkins-custom build check and use `-z` to reference service accounts

### CODE REVIEW

File: ansible/roles/jenkins/tasks/main.yml [hunk 1]
```diff
@@ -42,14 +42,11 @@
   shell: "{{ oc }} process -f {{ role_path }}/files/base-image-pipeline.yaml --param BASE_IMAGE_TAG='nodejs:6' --param APP_BC='nodejs-mongo-persistent' --param APP_NAME='nodejs-mongo-persistent' --param EMAIL_LIST='aweiteka@redhat.com jcallen@redhat.com' --param=APP_PIPELINE='lifecycle-app-lifecycle' -n {{ project_name }} | {{ oc }} apply --force=true -f - -n {{ project_name }}"
 - name: Create Jenkins base image pipeline
   shell: "{{ oc }} process -f {{ role_path }}/files/base-image-pipeline.yaml -n {{ project_name }} | {{ oc }} apply --force=true -f - -n {{ project_name }}"
-- name: Get last jenkins-custom build number
-  command: "{{ oc }} get bc jenkins-custom --template '{{ '{{' }} .status.lastVersion {{ '}}' }}'"
-  register: jenkins_custom_build_number
 - name: Get last jenkins-custom build status
-  command: "{{ oc }} get build jenkins-custom-{{ jenkins_custom_build_number.stdout }} --template '{{ '{{' }} .status.phase {{ '}}' }}'"
+  command: "{{ oc }} get build jenkins-custom-1 -n {{ project_name }} --template '{{ '{{' }} .status.phase {{ '}}' }}'"
```
Comment (aweiteka) [inline ansible/roles/jenkins/tasks/main.yml]: I was trying to avoid hard-coding the build number

File: ansible/roles/pusher/tasks/main.yml [hunk 2]
```diff
@@ -9,17 +9,3 @@
 - name: Link docker push central registry secret
   command: "{{ oc }} secrets link builder push-external-registry -n {{ project_name }}"
   when: central_registry_secret|failed
-- name: Check if push prod registry secret exists
-  command: "{{ oc }} get secret push-prod-registry -n {{ project_name }}"
-  register: prod_registry_secret
-  ignore_errors: true
-- name: Docker push prod registry secret
-  command: "{{ oc }} secrets new-dockercfg push-prod-registry --docker-username=unused --docker-password={{ hostvars['prod']['pushtoken'] }} --docker-email=unused -n {{ project_name }}"
```
Comment (jcpowermac) [inline ansible/roles/pusher/tasks/main.yml:17]: @codificat there is a buildconfig that outputs to the external registry wouldn't this still be needed?

File: ansible/roles/jenkins/tasks/main.yml (hunk 1 above)
Comment (jcpowermac) [inline ansible/roles/jenkins/tasks/main.yml]: @codificat What if the build number is something other than one?

Comment (codificat) [reply ansible/roles/jenkins/tasks/main.yml]: Right, I was thinking only on initial run, when the buildConfig is created - at which point the first build will obviously be build 1... but certainly it won't be when this runs again.

On the other hand: what triggered this change was a failure of the playbook due to a race: reaching that check before any build was triggered - resulting in a value of `0` in the build number check.

Bottom line: this part needs some improvement, will re-review.

Comment (aweiteka) [reply ansible/roles/jenkins/tasks/main.yml]: FWIW, the goal is to be able to run ansible any time without arbitrarily starting builds.

I know what we had wasn't working every time. thanks for looking at it.

File: ansible/roles/pusher/tasks/main.yml (hunk 2 above)
Comment (codificat) [reply ansible/roles/pusher/tasks/main.yml:17]: @jcpowermac AFAICT output goes to the `central_registry` but `prod_registry` is not actually used/referenced... so the suggestion here is to remove it.... did I miss something?

Comment (jcpowermac) [reply ansible/roles/pusher/tasks/main.yml:17]: @codificat it would have helped if I read all the tasks.  Were good.

**Pull request merged**